pwa-backend/src/main/java/io/growingabit/backoffice/controllers/InvitationController.java

```diff
@@ -9,35 +15,31 @@
 import javax.ws.rs.Produces;
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
-
-import com.googlecode.objectify.SaveException;
-
-import io.growingabit.backoffice.dao.InvitationDao;
-import io.growingabit.backoffice.model.Invitation;
-import io.growingabit.jersey.annotations.Secured;
-import io.growingabit.jersey.utils.UserRoles;
+import org.apache.commons.lang3.exception.ExceptionUtils;
 
 @Path("backoffice/invitation")
 @Secured
 @RolesAllowed(UserRoles.ADMIN)
 public class InvitationController {
 
-  private InvitationDao dao = new InvitationDao();
+  private final Logger logger = Logger.getLogger(InvitationController.class.getName());
+  private final InvitationDao dao = new InvitationDao();
 
   @GET
   @Produces(MediaType.APPLICATION_JSON)
   public Response list() {
-    return Response.ok(dao.findAll()).build();
+    return Response.ok(this.dao.findAll()).build();
   }
 
   @POST
   @Consumes(MediaType.APPLICATION_JSON)
   @Produces(MediaType.APPLICATION_JSON)
-  public Response save(Invitation invitation) {
+  public Response save(final Invitation invitation) {
```

Comment:
Se lo metti final, Ofy e poi in grado di scriverci l'id?

Reply:
No, final non impedisce di modificare l'oggetto. Impedisce di cambiare il puntatore. La variabile `invitation` punterà sempre a quanto passato nel metodo. Se provi a fare `invitation = new Invitation(....)` il compilatore ti da errore.